### PR TITLE
[le11] system-tools: migrate from p7zip to 7-zip and addon (2)

### PR DESCRIPTION
**Pull request by heitbaum**
- 7-zip: initial package
- system-tools: migrate from p7zip to 7-zip and addon (2)

### CODE REVIEW

File: packages/compress/7-zip/package.mk [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  fi
+}
+
+makeinstall_target() {
+  mkdir -p ${INSTALL}/usr/bin
+    if [ "${TARGET_ARCH}" = "aarch64" ]; then
+      cp -p ${PKG_BUILD}/.${TARGET_NAME}/CPP/7zip/Bundles/Alone2/b/g_arm64/7zz ${INSTALL}/usr/bin
+    else
+      cp -p ${PKG_BUILD}/.${TARGET_NAME}/CPP/7zip/Bundles/Alone2/_o/7zz ${INSTALL}/usr/bin
+    fi
```
Comment (CvH) [inline packages/compress/7-zip/package.mk:54]:
```suggestion
  mkdir -p ${INSTALL}/usr/bin
  if [ "${TARGET_ARCH}" = "aarch64" ]; then
    cp -p ${PKG_BUILD}/.${TARGET_NAME}/CPP/7zip/Bundles/Alone2/b/g_arm64/7zz ${INSTALL}/usr/bin
  else
    cp -p ${PKG_BUILD}/.${TARGET_NAME}/CPP/7zip/Bundles/Alone2/_o/7zz ${INSTALL}/usr/bin
  fi
```

File: packages/compress/7-zip/package.mk [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+make_host() {
+  # compile without 7-Zip's assembler code (not required in toolchain)
+  make CXX=${CXX} CC=${CC} -f makefile.gcc -C ${PKG_BUILD}/.${HOST_NAME}/CPP/7zip/Bundles/Alone
+}
+
+makeinstall_host() {
+  mkdir -p ${TOOLCHAIN}/bin
+    cp ${PKG_BUILD}/.${HOST_NAME}/CPP/7zip/Bundles/Alone/_o/7za ${TOOLCHAIN}/bin
```
Comment (CvH) [inline packages/compress/7-zip/package.mk:29]:
```suggestion
  cp ${PKG_BUILD}/.${HOST_NAME}/CPP/7zip/Bundles/Alone/_o/7za ${TOOLCHAIN}/bin
```

File: packages/compress/7-zip/package.mk [hunk 3]
```diff
@@ -0,0 +1,55 @@
+# SPDX-License-Identifier: GPL-2.0-only
+# Copyright (C) 2023-present Team LibreELEC (https://libreelec.tv)
+
+PKG_NAME="7-zip"
+PKG_VERSION="22.01"
+PKG_SHA256="393098730c70042392af808917e765945dc2437dee7aae3cfcc4966eb920fbc5"
+PKG_LICENSE="7-Zip"
+PKG_SITE="https://www.7-zip.org"
+PKG_URL="https://www.7-zip.org/a/7z${PKG_VERSION/.}-src.tar.xz"
```
Comment (CvH) [inline packages/compress/7-zip/package.mk:9]:
```suggestion
PKG_URL="https://www.7-zip.org/a/7z${PKG_VERSION/./}-src.tar.xz"
```

**Review by CvH — APPROVED** [review]
Just a friendly reminder that you cant use a variable name "7zip" because a variable isn't allowed to start with a number :) (not a problem at PR)

The other changes are reported by bash format check.

**Comment by heitbaum** [comment]
> Just a friendly reminder that you cant use a variable name "7zip" because a variable isn't allowed to start with a number :) (not a problem at PR)
> 
> The other changes are reported by bash format check.

Thanks. Merged 👍

**Pull request merged**